### PR TITLE
feat: allow specifying codec preferences on setTargetBitrate()

### PR DESCRIPTION
**Pull request by murillo128**
This PR allows to specify the codec preferences in order to choose the correct layer.  It will allow us to implement multi video codec simulcast and only use the layers that the client supports for layer switching.

### CODE REVIEW

File: lib/IncomingStreamTrack.js [hunk 1]
```diff
@@ -263,7 +266,8 @@ function getStatsFromIncomingSource(/** @type {Native.RTPIncomingSource} */ sour
 			totalBitrate	: layer.totalBitrate,
 			active		: layer.active, 
 			// provisional (set by updateStatsSimulcastIndex)
-			simulcastIdx: -1,
+			simulcastIdx	: -1,
+			codec		: "unknown"
```
Comment (bcostdolby) [inline lib/IncomingStreamTrack.js:270]: Do we have some kind of const we can use for this somewhere if we cant use null just to avoid typos in future maintainence?

File: lib/Transponder.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ const SemanticSDP	= require("semantic-sdp");
  * @property {boolean} [strict] If there is not a layer with a bitrate lower thatn target, stop sending media [Default: false]
  * @property {boolean} [smooth] When going to a lower simulcast layer, keep the higher one visible [Default: true]
  * @property {boolean} [useDefaultEncoding] If there are no active layers, use the default encoding [Default: false]
+ * @property {string[]} [codecs] Codec preferences list in ascending order, layers with codec not present in codec list will be ignored
```
Comment (bcostdolby) [inline lib/Transponder.js:29]: I think i misunderstood the meaning of "ascending" order here. The test looks like the most preferable is first in the list but ascending order makes me thing the most preferable would be the last in the list.

File: tests/transponder.js [hunk 3]
```diff
@@ -948,6 +948,164 @@ tap.test("Transponder::targetbitrate simulcast",async function(suite){
...
+			},
+	});
+
+	
+	await suite.test("top",async function(test){
+		try {
+			let bitrate;
+			//Target bitrate
+			const target = 1250000;
+			//Set bitrate
```
Comment (bcostdolby) [inline tests/transponder.js:1084]: Is testing codec preference of av1 vs h264 and "filtering" below but. think there are a few extra use cases that are missing. 

Like, to be sure it works probably also test ["h264", "av1"] order, but also interaction of priority with other types of ordering in the impl are missing. 

I found the code a little difficult to read in the impl with things like:
`layers = layers.sort((a,b) => codecSortByPreference(a,b) || ordering(a,b));
`

Is just a style i havent seen much before. To be sure it works correctly in various edge cases I think we are missing a bunch of tests.

File: lib/Transponder.js (hunk 2 above)
Comment (murillo128) [reply lib/Transponder.js:29]: yes, should be descending, good catch

File: tests/transponder.js (hunk 3 above)
Comment (murillo128) [reply tests/transponder.js:1084]: I would defer the js styles to @mildsunrise , but this is the the only way I found to easily concatenate filtering fucntions.

Comment (murillo128) [reply tests/transponder.js:1084]: `codecSortByPreference(a,b) || ordering(a,b)` will do first order based on codec preference and if they have same codec (i.e. the result is 0) then it will filter by `ordering` which is the svc layer traversal order

File: lib/IncomingStreamTrack.js (hunk 1 above)
Comment (murillo128) [reply lib/IncomingStreamTrack.js:270]: I would be tempted o just specifying `codec: undefined` here as it will later be populated by the `updateStatsSimulcastIndexAndCodec` but not sure if that would freak the typesystem
